diff --git a/autocomplete_light/registry.py b/autocomplete_light/registry.py
index b4d7dd9..d83ed33 100644
--- a/autocomplete_light/registry.py
+++ b/autocomplete_light/registry.py
@@ -178,12 +178,19 @@ class AutocompleteRegistry(dict):
 def _autodiscover(registry):
     """See documentation for autodiscover (without the underscore)"""
     import copy
+    from django import get_version
     from django.conf import settings
     from django.utils.importlib import import_module
     from django.utils.module_loading import module_has_submodule
 
-    for app in settings.INSTALLED_APPS:
-        mod = import_module(app)
+    if get_version() >= '1.7':
+        from django.apps import apps
+        apps = ((app.module, app.module.__name__)
+                for app in apps.app_configs.values())
+    else:
+        apps = ((import_module(app), app) for app in settings.INSTALLED_APPS)
+
+    for mod, app in apps:
         # Attempt to import the app's admin module.
         try:
             before_import_registry = copy.copy(registry)
